### ISSUE TITLE
Axios 1.3.x doesn't recognize noproxy config from .npmrc

### ISSUE THREAD

**Issue by ksathyanm** [axios/axios#5619]
### Describe the bug

I am behind corporate network. Axios 1.3.x uses the `proxy` config from `.npmrc` file but that proxy was setup only for corporate managed npm registry. As a workaround, I tried adding `noproxy` to `.npmrc` file and it didn't work either.

### Expected behavior

If Axios automatically uses `proxy` from `npmrc` file, it should respect `noproxy` config too.

### Axios Version

1.3.x

### Adapter Version

HTTP

### Node.js Version

Node 18 / npm 9

**Comment by npiyush97** [comment]
noproxy=example.com
so you tried ?

**Comment by ksathyanm** [comment]
@npiyush97 Yes, looks like `proxy-from-env` doesn't support `noproxy`

**Comment by ksathyanm** [comment]
@mbargiel @jasonsaayman Can you please share your thoughts on an issue with `proxy-from-env` dependency. Looks like @Rob--W is occupied with other stuff and not sure if he will be able to provide his support on this PR https://github.com/Rob--W/proxy-from-env/pull/19.

**Comment by paztis** [comment]
Just understood the defect

in the .npmrc the correct spelling is noproxy and not no-proxy
It creates the environment variable npm_config_noproxy and not the env var npm_config_no_proxy

IN axios you;re looking at the bad environment variable.
For example if I odify my .npmrc and rename noproxy in no-proxy axios works correctly, but all my other npm services are no more working.

**Comment by paztis** [comment]
ok this is what is explained into this defect ... :)

**Comment by slin-sap** [comment]
Hi, I also encountered this issue on axios 1.6.1. Setting noproxy in .npmrc doesn't work. Setting environment variable  NO_PROXY is a temporary workaround. May I ask what is the current status of this issue?

**Comment by Rob--W** [comment]
> Hi, I also encountered this issue on axios 1.6.1. Setting noproxy in .npmrc doesn't work. Setting environment variable NO_PROXY is a temporary workaround. May I ask what is the current status of this issue?

Oops, this fell off my radar. I've set a reminder for myself to look at publishing a `proxy-from-env` update this weekend.

**Comment by KrayzeeKev** [comment]
The issue here is that proxy-from-env looks at npm_config_http_proxy and npm_config_https_proxy
WHY? That's ridiculous. We're not running npm, we're running our software. The npm config proxies are for npm to get its modules. Axios shouldn't be using npm configs for doing non-npm things. 
I doubt proxy-from-env is going to change (there's zero activity there for over a year) but I think Axios should seriously consider not using it. Go back to manually checking the http_proxy, https_proxy and no_proxy environment variables. That's it.
noproxy is NOT the solution here - I can't list EVERY host I use in the noproxy just because I want npm to talk to nexus using a proxy.
The contents of ~/.npmrc should have ZERO effect on any software other than npm. Certainly not MY software that uses Axios. This took me an entire day to debug and now I have to break npm to make my software work.

**Comment by Rob--W** [comment]
> The issue here is that proxy-from-env looks at npm_config_http_proxy and npm_config_https_proxy
> WHY?

Because people wanted a way to rely on npm config, in https://github.com/Rob--W/proxy-from-env/issues/8

In retrospect that should have been opt-in.

> The contents of ~/.npmrc should have ZERO effect on any software other than npm. Certainly not MY software that uses Axios.

I agree. I'm going to work on a 2.0.0 release that doesn't look at npm vars by default, but that exports a method called `getProxyFromEnvAndNpmConfig` for those who really want to read from the npm config for some reason. I'm going to track that work at https://github.com/Rob--W/proxy-from-env/issues/13

**Comment by KrayzeeKev** [comment]
Awesome. Perhaps maybe something even more generic where you can provide an arbitrary prefix or prefixes for the variables so you can call it with "npm_config_" but you could also call it with "curl_" or whatever else.

**Comment by bc2009** [comment]
> The issue here is that proxy-from-env looks at npm_config_http_proxy and npm_config_https_proxy
> 
> WHY? That's ridiculous. We're not running npm, we're running our software. The npm config proxies are for npm to get its modules. Axios shouldn't be using npm configs for doing non-npm things. 
> 
> I doubt proxy-from-env is going to change (there's zero activity there for over a year) but I think Axios should seriously consider not using it. Go back to manually checking the http_proxy, https_proxy and no_proxy environment variables. That's it.
> 
> noproxy is NOT the solution here - I can't list EVERY host I use in the noproxy just because I want npm to talk to nexus using a proxy.
> 
> The contents of ~/.npmrc should have ZERO effect on any software other than npm. Certainly not MY software that uses Axios. This took me an entire day to debug and now I have to break npm to make my software work.

agreed. this baffled me for months until out of fortune someone told me about this. i think it is reasonable to assume process launched from npm scripts to respect proxy environment variables like `http_proxy` for `python` or `curl`. But it's really a surprise to learn .npmrc proxy settings affect a nodejs process.

**Comment by jasonsaayman** [comment]
based on reading this issue we cannot solve this in Axios so i will close this with a note to add this to docs / recipes / known gotchas'
